src/logic/wallet.ts

```diff
@@ -0,0 +1,75 @@
+import { AsyncStorage } from 'react-native'
+import { BehaviorSubject, Subject, Observable } from 'rxjs'
+import * as util from 'ethereumjs-util'
+import * as keychain from 'react-native-keychain'
+
+import * as T from '../typings'
+
+const STORAGE_KEY = 'wallet-public-info'
+
+const accountsSub = new BehaviorSubject<T.EthAccount[] | undefined>(undefined)
+
+const loadAccounts = () =>
+  Observable.defer(() => AsyncStorage.getItem(STORAGE_KEY))
+    .map(acc => JSON.parse(acc || '[]'))
+    .do(acc => accountsSub.next(acc))
+    .do(acc => console.log('ACCOUNTS', acc))
+    .retryWhen(errs => errs.delay(1000))
+
+export const accounts = accountsSub.asObservable()
+
+// todo: should not be needed
+// export const getAccountsSync = () => accountsSub.value || []
+
+// todo: discuss
+export const addAccount = (privateKey: string): Promise<boolean> => {
+  if (util.isValidPrivate(util.toBuffer(privateKey))) {
+    const address = util.bufferToHex(util.privateToAddress(privateKey))
+    const current = accountsSub.value || []
+    if (current.find(a => a.address === address)) {
+      return Promise.resolve(false)
+    }
+    const accounts = current.concat([{
+      address,
+      amount: 0 // todo: monitoring
+    }])
+    return AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(accounts))
+      .then(() => keychain.setGenericPassword('', privateKey, {
```

Comment:
Maybe we could use async/await? It makes the code more readable

Reply:
We could, but then we would need to wrap the code in a `try catch` block. Another thing is that promise chains are easier upgradable to Observable than `async/await` - e.g. if we wanted to add some retry mechanism.
But sure feel free to use `async/await`, whenever it makes sense to you.